doc/source/loggers.rst

```diff
@@ -13,26 +13,133 @@ by caproto, below. Because recalling the name is not always convenient, certain
+    [D 15:19:14.135          client:  454] [CLIENT] [0.0.0.0:0] --->>> [10.2.255.255:5065] Sending 16 bytes to 10.2.255.255:5065
+    [D 15:19:14.136          client:  832] [CLIENT] Broadcaster search-retry thread has started.
+    [D 15:19:14.136          client: 1079] [CLIENT] Context search-results processing loop has started.
+    [D 15:19:14.137          client:  872] [CLIENT] Sending 6 SearchRequests
+
+To add logstash handler, you can submit logs to a centralized log database:
+
+.. code-block:: python
+
+    import logstash
+    logstash_handler = logstash.TCPLogstashHandler(<host>, <port>, version=1)
+    logger.addHandler(logstash_handler)
+
+To add a file handler:
+
+.. code-block:: python
+
+    file_handler = logging.FileHandler('caproto.log')
+    file_handler.setLevel('DEBUG')
+    logger.addHandler(file_filter)
+
+To add a filter:
+
+.. code-block:: python
+
+    from caproto._log import PVFilter, AddressFilter, RoleFilter
+    std_handler.addFilter(PVFilter(['complex:*', 'simple:B'], level='DEBUG', exclusive=False))
+    std_handler.addFilter(AddressFilter(['10.2.227.105']))
+    std_handler.addFilter(RoleFilter('Client', level = 'DEBUG'))
+
+For example, ``PVFilter(['complex:*', 'simple:B'], level='DEBUG', exclusive=False)`` meaning
+You want to block any message below level=DEBUG level except it related complex:* or simple:B and
+keep the env, config or misc message by exclusive=False
 
 See :ref:`threading_loggers` for more handy examples.
 
+Logger's level vs handler's level
+=================================
+
+Make sure you understand how level control loggging in logger and handler.
+Both has method setLevel(...) which allow you do logger.setLevel(...) or handler.setLevel(...).
+You may only have one logger in your package. But, mutiple handlers could be added to one logger.
+Logger's level influence all handler. So the effective level is the intersection of logger's level
+and handler's level.
+
+
+Here INFO will effect
+
+.. code-block:: python
+
+    import logging
+    logger = logging.getLogger('caproto')
+    logger.setLevel('DEBUG')
+    handler = logging.StreamHandler()
+    handler.setLevel('INFO')
+    logger.addHandler(handler)
+    logger.debug('This is debug message')
+    logger.info('This is info message')
+    logger.warning('This is warn message')
+
+.. code-block:: python
+
+    This is info message
+    This is warn message
+
+Here WARNING will effect
+
+.. code-block:: python
+
+    import logging
+    logger = logging.getLogger('caproto')
+    logger.setLevel('WARNING')
+    handler = logging.StreamHandler()
+    handler.setLevel('INFO')
+    logger.addHandler(handler)
+    logger.debug('This is debug message')
+    logger.info('This is info message')
+    logger.warning('This is warn message')
+
+.. code-block:: python
+
+    This is warn message
+
+To avoid confused level setting, we recommend leave logger's level 'NOTSET' and use
```

Comment:
This sort of 'warning' should be in a big box saying "DON'T DO THIS! DO THIS INSTEAD! And here are the reasons why: ..."